collector/library/pod.js

```diff
@@ -156,46 +154,72 @@ class Pod {
-            try {
-                const request = await fetch(`${this.targetUrl}/batch/${batch.id}?follow`);
-                const bytesAlreadySentToLogstashFromThisBatch = await this.getBytesSentFromThisParticularBatch({ batch });
-                let bytesSentAccumlator = 0;
-
-                request.body.on('data', async (data) => {
-                    let sendData = false;
-                    if (bytesAlreadySentToLogstashFromThisBatch > 0 && bytesSentAccumlator >= bytesAlreadySentToLogstashFromThisBatch) {
-                        sendData = true;
-                    }
+    async handleBatch(batch) {
+        console.log(`${chalk.greenBright('[NEW BATCH]')} (id: ${batch.id})`);
+        const request = await fetch(`${this.targetUrl}/batch/${batch.id}?follow`);
 
-                    if (bytesAlreadySentToLogstashFromThisBatch === 0) {
-                        sendData = true;
-                    }
+        const bytesAlreadySentToLogstashFromThisBatch = await this.getBytesSentFromThisParticularBatch({ batch });
+        let bytesSentAccumlator = 0;
 
-                    if (sendData) {
-                        await this.submitDataToLogstash({ data, batch });
-                    }
+        if (!request.ok) {
+            throw new Error('The request did not finish successfully!');
+        }
 
-                    bytesSentAccumlator += data.length;
-                });
+        await new Promise((res, rej) => {
+            this.em.on(DEAD_POD, () => {
+                rej(err);
+                request.body.destroy();
+            });
 
-            } catch (err) {
-                // The request got an error for some reason,
-                // The next round for new files scouting will pick up where we left off ;-)
-            }
-        });
+            request.body.on('data', async (data) => {
+                if (this.dead === true) { // Our pod has stopped, kill this request
+                    rej(err);
+                    request.body.destroy();
+                }
+
+                let sendData = false;
+                if (bytesAlreadySentToLogstashFromThisBatch > 0 && bytesSentAccumlator >= bytesAlreadySentToLogstashFromThisBatch) {
+                    sendData = true;
+                }
+
+                if (bytesAlreadySentToLogstashFromThisBatch === 0) {
+                    sendData = true;
+                }
+
+                if (sendData) {
+                    await this.submitDataToLogstash({ data, batch }).catch((err) => {
+                        rej(err);
+                        request.body.destroy();
+                    });
+                }
+
+                bytesSentAccumlator += data.length;
+            });
 
-        this.em.on(UNACKED_DATA, async ({ data, batch, packetId }) => {
-            await this.submitDataToLogstash({ data, batch, packetId });
+            request.body.on('error', (err) => {
+                console.log(err);
```

Comment:
I know I said before that we might want to ignore errors but now I don't see the problem with killing the request object and throwing an exception